### DISCUSSION TITLE
Native nested function chain for neasted type  in SELECT Items

### DISCUSSION THREAD

**Discussion by douenergy** [duckdb/duckdb#6717] [Ideas]
Given a relation with a column containing lists, i want to count every row for what is  the number of elements in the list is greater than 40.
|                 numbers                  |
|--------------------------------------|
| [46, 70, 76, 77, 78, 80]             |
| [43, 46, 47, 53, 57, 64, 74, 77]     |
| [41, 51, 53, 54, 59, 70, 71]         |
| [44, 45, 47, 53, 57, 59, 60, 66, 77] |
| [70, 71, 75, 76, 77, 78]             |

this works really well 😀
```sql 
SELECT list_aggregate( list_filter( numbers,  x -> x > 40 ) , 'count') as greater_than_40s 
FROM relation
```

If we support the following syntax  , I think will be really awesome 🥳

```sql 
SELECT numbers.filter( x -> x > 40).aggregate('count') as greater_than_40s 
FROM relation
```
The idea is to enable function chaining for nested data types such as **lists**, **maps**, and **structs**, in order to perform operations on them more readable(IMHO).

**Comment by Mytherin** [comment]
Thanks for the suggestion! I agree that the dot syntax looks a lot cleaner. I have made an implementation of this in #6725.
